bundles/SeoBundle/src/Installer.php

```diff
@@ -0,0 +1,63 @@
+<?php
+declare(strict_types=1);
+/**
+ * Pimcore
+ *
+ * This source file is available under two different licenses:
+ * - GNU General Public License version 3 (GPLv3)
+ * - Pimcore Commercial License (PCL)
+ * Full copyright and license information is available in
+ * LICENSE.md which is distributed with this source code.
+ *
+ *  @copyright  Copyright (c) Pimcore GmbH (http://www.pimcore.org)
+ *  @license    http://www.pimcore.org/license     GPLv3 and PCL
+ */
+
+namespace Pimcore\Bundle\SeoBundle;
+
+use Pimcore\Extension\Bundle\Installer\SettingsStoreAwareInstaller;
+
+class Installer extends SettingsStoreAwareInstaller
+{
+    protected const USER_PERMISSIONS = [
+        'robots.txt',
+        'seo_document_editor',
+        'http_errors',
+    ];
+
+    public function install(): void
+    {
+
+        $this->addUserPermission();
+        parent::install();
+    }
+
+    public function uninstall(): void
+    {
+
+        $this->removeUserPermission();
+        parent::uninstall();
+    }
+
+    private function addUserPermission(): void
+    {
+        $db = \Pimcore\Db::get();
+
+        foreach (self::USER_PERMISSIONS as $permission) {
+            $db->insert('users_permission_definitions', [
```

Comment:
I moved the migration to the CoreBundle. It should by default get marked as installed, so the permission should not be installed again. If you do a fresh install, this problem should not occur, because I removed the permissions in the InstallerBundle. So basically when you upgrade, the bundle should be marked as installed by default. 

Wdyt?

Reply:
sounds good to me, at least for a seamless upgrade from 10.x, but not sure about from a skeleton point of view, if anything "optional" should be deactiveted by default. Let's ask in the daily